### ISSUE TITLE
Fix Request: InconsistentEqualsAndHashCode lacks support for use of 'this' 

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#90]
False positive:
```
class EqHashTryoutThis2 {
    String field1;
    String field2;

    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        EqHashTryout2 that = (EqHashTryout2) o;
        return Objects.equals(this.field1, that.field1) &&
                Objects.equals(this.field2, that.field2);
    }
    public int hashCode() {
        return Objects.hash(field1, field2);
    }
}
```
